### DISCUSSION TITLE
Highlighting current "on this page" section in Docs

### DISCUSSION THREAD

**Discussion by dobaret** [thuliteio/doks#589] [Q&A]
Hello! 

Very non-technical user here: in Docs, how could the current "On the page" section be highlighted, like the current page is in the left sidebar?

It would be quite useful in large pages.

(Btw, thank you @h-enk for Doks, it's amazing!)

**Comment by h-enk** [comment]
That would require a little JavaScript, by the first looks of it. [Tocbot](http://tscanlin.github.io/tocbot/) seems like a good candidate for this.

I will be considering making this an option in Doks.

**Reply by dobaret** [reply]
Okay, thank you for your answer!

**Comment by nk9** [comment]
I'd like this feature as well, but it works best if you have headers with lots of content beneath.  Otherwise, figuring out which TOC item is the "current" one can get difficult toward the end of the page. As you can see on the Tocbot example site, they solve the problem by having a ton of extra space at the bottom of the page. Feels kind of hacky. Another thing to be careful of is to make sure that the scroll position takes account of any header/menu bar on the page (Tocbot seems to do this).

**Reply by h-enk** [reply]
Thanks, yeah, will need some more looking into it. Maybe a better alternative is out there?
